File: ExpenseSplitter.Frontend/ExpenseSplitter.Web/src/app/models/expense/expense-update.model.ts
import { ExpenseTypeEnum } from 'src/app/models/expense/expense-type.enum';
import { ExpensePartModel } from './expense-part.model';

export class ExpenseUpdateModel {
    id?: number;
    name: string;
    type: ExpenseTypeEnum;
    paidAt: Date;
    payerId: number;

    parts: ExpensePartModel[];
}

File: ExpenseSplitter.Backend/ExpenseSplitter.Api/Models/Expenses/ExpenseUpdateModel.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using ExpenseSplitter.Api.Data;
using ExpenseSplitter.Api.Infrastructure;

namespace ExpenseSplitter.Api.Models.Expenses
{
    public class ExpenseUpdateModel
    {
        public int? Id { get; set; }

        [Required, MinLength(Constants.ExpenseNameMinLength), MaxLength(Constants.ExpenseNameMaxLength)]
        public string Name { get; set; }

        [Required]
        public ExpenseType Type { get; set; }

        [Required]
        public DateTime PaidAt { get; set; }

        [Required]
        public int PayerId { get; set; }

        [Required, MinLength(1)]
        public List<ExpensePartModel> Parts { get; set; }
    }
}
